Updates to SNow customizations PR (#616)
hashicorp/terraform-docs-common: website/docs/cloud-docs/integrations/service-now/service-catalog-terraform/example-customizations.mdx

The new item is now available in the Terraform Service Catalog. To make the
new item accessible to your end users via the Service Portal, follow these
steps:
1. Navigate to the configuration page of the item you want to make available as
   described in the previous section.
1. Locate the "Catalogs" field on the configuration page and click on the lock
1. Navigate to the configuration page of the item you want to make available.
1. Locate the **Catalogs** field on the configuration page and click the lock
   icon next to it.
1. In the search bar, type "Service Catalog" and select it from the search
   results. Add "Service Catalog" to the list of catalogs associated with the
   item. Click on the lock icon again to lock the changes.
   item. Click the lock icon again to lock the changes.
   ![Screenshot: ServiceNow Enable Service Portal](/img/docs/servicenow-catalog-service-portal.png "Screenshot of adding the Catalog Item to the Service Portal")
1. Finally, click the "Update" button at the top of the page to save the
modifications. After completing these steps, your end users will be able to
access the new item through the Service Portal of your ServiceNow instance. The

> Service Portal can be accessed by navigating to All > Service Portal Home.

Reviewer: Please keep the path to the Service Portal. Many customers really don't know what it is or how to get there.

Reviewer: `The Service Portal can be accessed by navigating to All > Service Portal Home.` <<< there's a lot of confusion about this even among our own SMEs. I get questions about this in Slack every month. Please bring back this sentence.

      1. Add your custom variables by clicking the **Create Input** button. Ensure that the variable names match your Catalog Item variables and select the variable type that matches each variable. Click **Save**.
      ![Screenshot: ServiceNow Add Variables To Action](/img/docs/servicenow-catalog-add-variables-to-action.png "Screenshot of adding ServiceNow Action Input Variables")
      1. Open the **Script step** within the Action. Remove any example variables
         and add your custom variables by clicking **Create Variable** at the
         bottom. Enter the name of each variable and drag the corresponding data
         pill from the right into the **Value field**.
         ![Screenshot: ServiceNow Add Script Step Variables To Action](/img/docs/servicenow-catalog-adjust-script-variables.png "Screenshot of adjusting ServiceNow Action Script Variables")
      1. Click **Save** and then **Publish**.
1. Reopen the Flow and attach the newly created Action to the Flow
   after "Get Catalog Variables" step:

>    1. Remove the previous Action that you copied earlier and replace it with

Reviewer: ```suggestion
   1. Remove the "Create Terraform Workspace with Vars" Action that you copied earlier and replace it with
```

      your newly created Action.
      ![Screenshot: ServiceNow Replace Action Step](/img/docs/servicenow-catalog-replace-action.png "Screenshot of replacing ServiceNow Action step")
   1. Connect the new Action to the Flow by dragging and dropping the data pills
      from the "Get Catalog Variables" Action to the corresponding inputs of
      your new Action. Click **Done** to save this step.
      ![Screenshot: ServiceNow Fill Variables for Action](/img/docs/servicenow-catalog-fill-new-action-step.png "Screenshot of filling out ServiceNow Action variables")
   1. Click **Save**.

>    1. Click **Activate**.

Reviewer: Now that you removed the brief context around these steps, it's not clear what's the difference between Save and Activate.

# Example Customizations

This example use case creates a Terraform Catalog Item for requesting resources
with custom variable values passed to the Terraform configuration.

## Make a copy of the existing Catalog Item

The application provides pre-configured [Catalog
Items](/terraform/cloud-docs/integrations/service-now/service-catalog-terraform/developer-reference#example-service-catalog-flows-and-actions)
for immediate use. We suggest you create a copy of the most recent version of the
Catalog Item to ensure you have access to the latest features and
improvements. Make a copy of the most appropriate Catalog Item for your specific
business requirements by following these steps:

1. Navigate to **All > Service Catalog > Catalogs > Terraform Catalog**, review the available
   Catalog Items and identify those based on Flows, which are denoted by the
   suffix "Flow" in their names. When you select a Catalog Item, we

>    recommended you choose those based on Flows rather than Workflows. 

Reviewer: ```suggestion
   recommend you choose those based on Flows rather than Workflows. 
```

1. Access the **Process Engine** tab in the Catalog Item configuration. Take
   note of the Flow name associated with the Catalog Item, as you will need to
   create a copy of this Flow as well.
   ![Screenshot: ServiceNow Process Engine](/img/docs/servicenow-catalog-process-engine.png "Screenshot of the ServiceNow Configure Catalog Item – Process Engine tab")
1. Start the copying process: 
   1. Click the **Copy** button above the **Related Links** section. 
   1. Assign a new name to the copied Catalog Item. 
   1. Optionally, modify the description and short description fields. 
      Right-click the header and select **Save**.
   ![Screenshot: ServiceNow Copy Item](/img/docs/servicenow-catalog-copied-item.png "Screenshot of the copied ServiceNow Catalog Item")

## Adjust the Variable Set

If a Catalog Item requires users to input variable values,
you must update the variable set with the required variables.
Although some default Catalog Items come with pre-defined example variables, it
is common practice to remove these and replace them with your own custom
variables.

1. Create a new Variable Set.
   1. On the Catalog Item's configuration page, under the **Related Links**

>       section, click the Variable Sets tab.

Reviewer: ```suggestion
      section, click the **Variable Sets** tab.
```

   1. Click the **New** button to create a new variable set. Ensure that the
   variables in your new set match the variables required by your Terraform
   configuration.
   ![Screenshot: ServiceNow New Variable Set](/img/docs/servicenow-catalog-new-varset.png)
   1. Select "Single-Row Variable Set" and provide a title and description.
   1. Click **Submit**. Upon submission, you will be redirected back to the Catalog
   Item's configuration page.
   ![Screenshot: ServiceNow New Variable Set Form](/img/docs/servicenow-catalog-new-varset-form.png "Screenshot of the ServiceNow Catalog Item – new Variable Set")
   1. Click the name of your newly created Variable Set and create your 
   variables. You must follow the [naming convention for Terraform
   variables](/terraform/cloud-docs/integrations/service-now/service-catalog-terraform/developer-reference#terraform-variables-and-servicenow-variable-sets).

>    ServiceNow offers various types of variable representation (ssuch as trings,

Reviewer: ```suggestion
   ServiceNow offers various types of variable representation (such as strings,
```

   booleans, and dropdown menus). Refer to the [ServiceNow documentation on
   variables](https://docs.servicenow.com/csh?topicname=c_ServiceCatalogVariables.html&version=latest)
   and select the types that best suit your use case. You can also set default

>    values for the variables in the **Default Value** tab, which ServiceNow will prefilled

Reviewer: ```suggestion
   values for the variables in the **Default Value** tab, which ServiceNow will prefill
```

>    for the end URLSearchParams.

Reviewer: ```suggestion
   for the end users in the request form.
```

   and click the "Edit" buttton. 
   and click the **Edit** button. 
   1. Move the "Workspace Variables" Set from the right side to the left side
   and click "Save". This action will remove the example variables from the
   form, leaving only your custom variables. You should not remove the
   "Workspace Request Create" or the "Workspace Request Update" Sets, as they
   contain mandatory variables and should not be modified.
   and click **Save**. Do not remove the
   "Workspace Request Create" or the "Workspace Request Update" Sets.
   ![Screenshot: ServiceNow Remove Example Variables](/img/docs/servicenow-catalog-remove-example-variables.png "Screenshot of the ServiceNow Catalog Item – new variables")

### Make a copy of the Flow and Action
## Make a copy of the Flow and Action

1. In your ServiceNow instance, access the Flow Designer using one of the
   following methods:
   1. Open the ServiceNow Studio through All > Studio and navigate to the
1. Open the ServiceNow Studio by navigating to **All > Studio** and open the
      "Terraform" application. Once in the "Terraform" application, navigate to
      "Flow Designer" > "Flows".

>    1. Another way to access the same interface is to type "Flow Designer" in the

Author: thanks! That makes sense, I'm adding a commit now to add this back

>          ![Screenshot: ServiceNow Add Variables To Action](/img/docs/servicenow-catalog-add-variables-to-action.png "Screenshot of adding ServiceNow Action Input Variables")

Reviewer 2: I think we can also remove this one:
```suggestion
```

>    1. Remove the previous Action that you copied earlier and replace it with

Reviewer 2: This sentence is kind of confusing, I think the user will need to know exactly which action to copy and replace here with either an example name or a link to a previous step

Reviewer: There's a screenshot visualizing this step below, if it helps :)

Reviewer 2: We try not to rely on screenshots to bridge word gaps for folks to ensure we are prioritizing accessible and understandable language for those using screenreaders. 

@jastr945 I think this is referencing the second action, correct?

```suggestion
   1. Remove the second Action that you previously copied earlier and replace it with
```

Reviewer: Let's reference it by name ("Create Terraform Workspace with Vars") - added a suggestion.

>    ![Screenshot: ServiceNow Update Process Engine](/img/docs/servicenow-catalog-update-process-engine.png "Screenshot of updating Process Engine for the Catalog Item")

Reviewer 2: Another one we can remove
```suggestion
```
